pertpy/plot/_plotperturbscore.py

```diff
@@ -0,0 +1,163 @@
+from __future__ import annotations
+
+import copy
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from plotnine import (
+    aes,
+    element_blank,
+    element_text,
+    facet_wrap,
+    geom_density,
+    geom_point,
+    ggplot,
+    scale_color_manual,
+    theme,
+    theme_classic,
+    xlab,
+    ylab,
+)
+
+
+def plotperturbscore(
+    adata: AnnData,
+    labels: str,
+    target_gene: str,
+    mixscape_class="mixscape_class",
+    color="orange",
+    split_by: str = None,
+    before_mixscape=False,
+    perturbation_type: str = "KO",
+):
+    """Density plots to visualize perturbation scores calculated from RunMixscape function.
+    https://satijalab.org/seurat/reference/plotperturbscore
+
+    Args:
+        adata: The annotated data object.
+        labels: The column of `.obs` with target gene labels.
+        target_gene: Target gene name to visualize perturbation scores for.
+        mixscape_class: The column of `.obs` with mixscape classifications.
+        color: Specify color of target gene class or knockout cell class. For control non-targeting and non-perturbed cells, colors are set to different shades of grey.
+        split_by: Provide the column `.obs` if multiple biological replicates exist to calculate
+            the perturbation signature for every replicate separately.
+        before_mixscape: Option to split densities based on mixscape classification (default) or original target gene classification. Default is set to NULL and plots cells by original class ID.
+        perturbation_type: specify type of CRISPR perturbation expected for labeling mixscape classifications. Default is KO.
+    Returns:
+        The ggplot object used for drawn.
+    """
+    perturbation_score = None
```

Comment:
Big blob of code afterwards. Maybe 1-2 sections + 1 line comments can help digest this?

Reply:
No problem